models/modules/unet_generator_attn/video_unet_generator_attn.py

```diff
@@ -0,0 +1,1391 @@
+                        self.cond_embed_dim,
+                        self.zero_dropout,
+                        out_channel=int(mult * self.inner_channel),
+                        use_checkpoint=use_checkpoint,
+                        use_scale_shift_norm=use_scale_shift_norm,
+                        norm=norm,
+                        efficient=efficient,
+                        freq_space=self.freq_space,
+                    )
+                ]
+                ch = int(mult * self.inner_channel)
+                if ds in attn_res:
+                    layers.append(
+                        MotionModule(
+                            in_channels=ch,
+                            num_attention_heads=8,
+                            num_transformer_block=2,
+                            attention_block_types=("Temporal_self", "Temporal_Self"),
+                            cross_frame_attention_mode=None,
+                            temporal_position_encoding=True,
+                            temporal_position_encoding_max_len=24,
+                            temporal_attention_dim_div=1,
+                            zero_initialize=True,
+                        )
+                    )
+                self.input_blocks.append(EmbedSequential(*layers))
+                self._feature_size += ch
+                input_block_chans.append(ch)
+            if level != len(channel_mults) - 1:
+                out_ch = ch
+                self.input_blocks.append(
+                    EmbedSequential(
+                        ResBlock(
+                            ch,
+                            self.cond_embed_dim,
+                            self.zero_dropout,
+                            out_channel=out_ch,
+                            use_checkpoint=use_checkpoint,
+                            use_scale_shift_norm=use_scale_shift_norm,
+                            down=True,
+                            norm=norm,
+                            efficient=efficient,
+                            freq_space=self.freq_space,
+                        )
+                        if resblock_updown
+                        else Downsample(
+                            ch,
+                            conv_resample,
+                            out_channel=out_ch,
+                            freq_space=self.freq_space,
+                        )
+                    )
+                )
+                ch = out_ch
+                input_block_chans.append(ch)
+                ds *= 2
+                self._feature_size += ch
+
+        self.middle_block = EmbedSequential(
+            ResBlock(
+                ch,
+                self.cond_embed_dim,
+                dropout,
+                use_checkpoint=use_checkpoint,
+                use_scale_shift_norm=use_scale_shift_norm,
+                norm=norm,
+                efficient=efficient,
+                freq_space=self.freq_space,
+            ),
+            MotionModule(
```

Comment:
Let's verify this because : 
- frame-only UNet has a "within-frame"  `AttentionBlock` here, that needs to be kept.
- I'm not sure the MM applies to the bottleneck : please double-check in publications an code.

Reply:
'AttentionBlock' is kept, 'MM' is added after it

Reply:
In the publication's code, whether MM is applied to the bottleneck depends on two options. However, in the two illustration figures in the publication, the bottleneck does not have MM.